iroha_client/src/client.rs

```diff
@@ -29,77 +28,75 @@ impl Client {
 
     /// Instructions API entry point. Submits one Iroha Special Instruction to `Iroha` peers.
     #[log]
-    pub async fn submit(&mut self, instruction: InstructionBox) -> Result<(), String> {
-        let network = Network::new(&self.torii_url);
+    pub fn submit(&mut self, instruction: InstructionBox) -> Result<(), String> {
         //TODO: specify account in the config or CLI params
-        let transaction = Transaction::new(
+        let transaction: Vec<u8> = Transaction::new(
             vec![instruction],
             AccountId::new("root", "global"),
             self.proposed_transaction_ttl_ms,
         )
-        .sign(&self.key_pair)?;
-        if let Response::InternalError = network
-            .send_request(Request::new(
-                uri::INSTRUCTIONS_URI.to_string(),
-                Vec::from(&transaction),
-            ))
-            .await
-            .map_err(|e| {
-                format!(
-                    "Error: {}, Failed to write a transaction request: {:?}",
-                    e, &transaction
-                )
-            })?
-        {
-            return Err("Server error.".to_string());
-        }
+        .sign(&self.key_pair)?
+        .into();
+        attohttpc::post(&format!(
```

Comment:
It's better to hide 3rd party http client under some iroha api.

Reply:
There are also more "1.0" libs like https://crates.io/crates/ureq - but it doesn't really matter what we will use in the end - just need to make it hidden in one place.